cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,229 @@ Cask Data Application Platform Release Notes
+  image) to use Ubuntu 16.04 LTS release.
+
+- :cask-issue:`CDAP-8990` - New configurations have been added to control the YARN
+  application master container memory size, maximum heap memory size, and maximum non-heap
+  memory size: twill.java.heap.memory.ratio, twill.yarn.am.memory.mb, and
+  twill.yarn.am.reserved.memory.mb.
+
+- :cask-issue:`CDAP-9003` - Increased the memory allocation for the CDAP Explore service
+  container to 2048MB.
+
+- :cask-issue:`CDAP-9017` - Simplified the status, next runtime of pipelines, total number
+  of running pipelines, and drafts in the pipeline list view UI.
+
+- :cask-issue:`CDAP-9070` - Added the ability to work with Data Preparation from within
+  the CDAP Studio UI. Users can now open a Wrangler transform, wrangle the data, and save it
+  back to the CDAP Studio.
+
+- :cask-issue:`CDAP-9077` - Added a "cdap apply-pack [pack]" command to the "cdap" script
+  that allows for upgrading of individual CDAP components.
+
+- :cask-issue:`CDAP-9242` - Improved the documentation of PartitionedFileSet properties.
+
+- :cask-issue:`CDAP-9278` - Added the running of HBase coprocessor upgrades concurrently
+  on CDAP Datasets.
+
+- :cask-issue:`CDAP-9282` - Made the AppMetadataStore and ScheduleStore compatible with an
+  older data format where the app version was not present, so that it is no longer necessary
+  to update these records to the new format in the Upgrade Tool.
+
+- :cask-issue:`CDAP-9283` - The migration of ScheduleStore and AppMetadataStore from the
+  app version-less format to the new format with a version will be performed by background
+  threads running in the CDAP Master process.
+
+- :cask-issue:`CDAP-9311` - Adding support for service discovery in plugins.
+
+- :cask-issue:`CDAP-9371` - Added convenience functions to set a Date in a
+  StructuredRecord as a long or a string.
+
+- :cask-issue:`HYDRATOR-1432` - Added a timer for realtime preview to stop a run
+  automatically. By default, it is one minute.
+
+Bug Fixes
+---------
+
+- :cask-issue:`CDAP-4213` - Fixed an issue in the CDAP startup script for Windows,
+  removing the requirement of being an admin to run the CDAP process.
+
+- :cask-issue:`CDAP-5715` - Made Plugin Endpoint invocation more robust. If a plugin's
+  parent can't instantiate the plugin necessary for invoking, CDAP will attempt with other
+  parents of the plugin and try to instantiate using them before retuning error.
+
+- :cask-issue:`CDAP-6348` - Fixed a Issue in Namespace delete which caused CDAP
+  Application's test cases to fail earlier in a Windows environment.
+
+- :cask-issue:`CDAP-7138` - Fixed a problem with duplicate log lines showing for a running
+  program.
+
+- :cask-issue:`CDAP-7884` - Fixed a bug that preview cannot read from datasets in real
+  space.
+
+- :cask-issue:`CDAP-8013` - Disabled the write-to-any-dataset for sinks when in preview.
+
+- :cask-issue:`CDAP-8396` - Added a "call-to-action" when the user had successfully
```

Comment:
calls to action were added in 4.1, in which we specifically called them out. Let's not have just this one here.